From bd1da7692120fbfb13a2295abeb9b20d69291296 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Revol?= <revol@free.fr>
Date: Sun, 17 Jan 2016 15:50:52 +0100
Subject: [PATCH] Xtranssock.c: Haiku has BSD UNIX socket but not SUN_LEN

(yet?)

The test seems quite incorrect and results in SUN_LEN as an undefined symbol on Haiku.
---
 Xtranssock.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Xtranssock.c b/Xtranssock.c
index d830e7c..f7c1637 100644
--- a/Xtranssock.c
+++ b/Xtranssock.c
@@ -1129,7 +1129,7 @@ TRANS(SocketUNIXCreateListener) (XtransConnInfo ciptr, const char *port,
     sockname.sun_len = strlen(sockname.sun_path);
 #endif
 
-#if defined(BSD44SOCKETS) || defined(SUN_LEN)
+#if defined(BSD44SOCKETS) && defined(SUN_LEN)
     namelen = SUN_LEN(&sockname);
 #else
     namelen = strlen(sockname.sun_path) + offsetof(struct sockaddr_un, sun_path);
@@ -2003,7 +2003,7 @@ TRANS(SocketUNIXConnect) (XtransConnInfo ciptr,
     sockname.sun_len = strlen (sockname.sun_path);
 #endif
 
-#if defined(BSD44SOCKETS) || defined(SUN_LEN)
+#if defined(BSD44SOCKETS) && defined(SUN_LEN)
     namelen = SUN_LEN (&sockname);
 #else
     namelen = strlen (sockname.sun_path) + offsetof(struct sockaddr_un, sun_path);
-- 
2.7.0

